lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -205,7 +206,7 @@ def store_content(resource)
       Puppet.info "Content for '#{resource[:source]}' already exists"
     else
       Puppet.info "Storing content for source '#{resource[:source]}'"
-      content = Puppet::FileServing::Content.indirection.find(resource[:source])
+      content = Puppet::FileServing::Content.indirection.find(resource[:source], {:environment => request.environment})
```

Comment:
I tried updating the static_compiler_spec, but failed... The tests don't actually work in large part, e.g.

```
    resource_catalog.resources.each do |resource|
          next unless resource.type == "File"
          expect(resource[:owner]).to eq(0)
          expect(resource[:group]).to eq(0)
          expect(resource[:mode]).to  eq(420)
        end
```

The above expectations are never reached because the `resource_catalog` doesn't contain any `File` resources as mentioned in https://github.com/puppetlabs/puppet/pull/4052#issuecomment-115178163.